### DISCUSSION TITLE
Help Compile DocumentServer.

### DISCUSSION THREAD

**Discussion by JonDoe224** [ONLYOFFICE/build_tools#535] [Q&A] [answered]
I complied DocumentServer on Ubuntu18.04 like "https://helpcenter.onlyoffice.com/installation/docs-community-compile.aspx?utm_source=blog&utm_medium=social&utm_campaign=build_tools"
I run command "./automate.py server", but this error appear like this..
========================================================
The following NEW packages will be installed:
  nodejs
0 upgraded, 1 newly installed, 0 to remove and 200 not upgraded.
Need to get 15.9 MB of archives.
After this operation, 80.5 MB of additional disk space will be used.
Err:1 https://deb.nodesource.com/node_10.x bionic/main amd64 nodejs amd64 10.24.1-1nodesource1
  Certificate verification failed: The certificate is NOT trusted. The revocation or OCSP data are old and have been superseded.  Could not handshake: Error in the certificate verification. [IP: 104.91.68.99 443]
E: Failed to fetch https://deb.nodesource.com/node_10.x/pool/main/n/nodejs/nodejs_10.24.1-1nodesource1_amd64.deb  Certificate verification failed: The certificate is NOT trusted. The revocation or OCSP data are old and have been superseded.  Could not handshake: Error in the certificate verification. [IP: 104.91.68.99 443]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
Error (sudo): 100
=========================================================

Please help me!

**Comment by ShockwaveNN** [comment]
Please respect others and format your question so it's not a mess to read

**Reply by JonDoe224** [reply]
Sorry, I have a ubuntu18.04, and compile it , show follow error:
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The following NEW packages will be installed:
nodejs
0 upgraded, 1 newly installed, 0 to remove and 200 not upgraded.
Need to get 15.9 MB of archives.
After this operation, 80.5 MB of additional disk space will be used.
Err:1 https://deb.nodesource.com/node_10.x bionic/main amd64 nodejs amd64 10.24.1-1nodesource1
Certificate verification failed: The certificate is NOT trusted. The revocation or OCSP data are old and have been superseded. Could not handshake: Error in the certificate verification. [IP: 104.91.68.99 443]
E: Failed to fetch https://deb.nodesource.com/node_10.x/pool/main/n/nodejs/nodejs_10.24.1-1nodesource1_amd64.deb Certificate verification failed: The certificate is NOT trusted. The revocation or OCSP data are old and have been superseded. Could not handshake: Error in the certificate verification. [IP: 104.91.68.99 443]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
Error (sudo): 100
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
How can resolve this problem?

**Reply by trofim24** [reply, marked as answer]
@JonDoe224 
The error you're seeing indicates a TLS/SSL certificate verification failure when trying to download Node.js from https://deb.nodesource.com.

This typically means that your system’s trusted certificate authorities (CAs) or certificate revocation data (CRLs/OCSP) are outdated or broken.

How to Fix It
1. Update Certificate Authorities
Make sure your system has the latest trusted certificates:
```
sudo apt-get update
sudo apt-get install --reinstall ca-certificates
```

2. Update Your System’s Time
Incorrect system time can cause certificate verification to fail:
```
timedatectl status
sudo timedatectl set-ntp true
```
